go/private/context.bzl

```diff
@@ -115,6 +115,7 @@ _COMPILER_OPTIONS_DENYLIST = dict({
 
 _LINKER_OPTIONS_DENYLIST = {
     "-Wl,--gc-sections": None,
+    "-pie": None, # See https://github.com/bazelbuild/rules_go/issues/3691.
```

Comment:
@fmeum This is the best I could come up with. The only other way seemed to add `supports_pic` to `_UNSUPPORTED_FEATURES` below but that results in a build failure with `--force_pic`, and does not do anything for `--linkopt=-pie`.